### PR TITLE
Hard fork extensions for `CombinationSlotState` and others

### PR DESCRIPTION
**Pull request by boscohyun**
- Hard fork extensions for `CombinationSlotState`
- Hard fork `RapidCombination`

### CODE REVIEW

**Review by area363 — APPROVED** [review]

**Review by sonohoshi — APPROVED** [review]

**Review by ipdae — CHANGES_REQUESTED** [review]
we missing test case for change.(hardfork case)

**Comment by sonohoshi** [comment]
> we missing test case for change.(hardfork case)

It means, we need to add a test case about model of `RapidCombination` 6 & 7? in `CombinationSlotStateExtensions`?

**Comment by ipdae** [comment]
@sonohoshi I was thinking of adding a scenario test.`RapidCombination` should succeed, `RapidCombination6` should fail.

**Comment by boscohyun** [comment]
@ipdae @sonohoshi I added some unit test. please review again.

**Review by sonohoshi — APPROVED** [review]

File: .Lib9c.Tests/Action/RapidCombinationTest.cs [hunk 1]
```diff
@@ -530,5 +531,192 @@ public void Execute_Throw_RequiredAppraiseBlockException()
...
         }
+
+        [Theory]
+        [InlineData(7, false)]
+        [InlineData(9, false)]
+        [InlineData(10, false)]
+        [InlineData(11, false)]
+        public void Execute_NotThrow_InvalidOperationException_When_TargetSlotCreatedBy(
+            int itemEnhancementResultModelNumber,
+            bool shouldThrow)
```
Comment (ipdae) [inline .Lib9c.Tests/Action/RapidCombinationTest.cs:542]: we can remove `shouldThrow`

File: .Lib9c.Tests/Action/RapidCombination6Test.cs [hunk 2]
```diff
@@ -0,0 +1,722 @@
...
+
+                default:
+                    break;
+            }
+
+            // NOTE: Do not update `mail`, because this test assumes that the `mail` was removed.
+            {
+                // var mail = new ItemEnhanceMail(resultModel, 0, random.GenerateRandomGuid(), requiredBlockIndex);
+                // avatarState.Update(mail);
+            }
```
Comment (ipdae) [inline .Lib9c.Tests/Action/RapidCombination6Test.cs:678]: is it intended?

File: .Lib9c.Tests/Action/RapidCombinationTest.cs [hunk 3]
```diff
@@ -530,5 +531,192 @@ public void Execute_Throw_RequiredAppraiseBlockException()
...
+
+                default:
+                    break;
+            }
+
+            // NOTE: Do not update `mail`, because this test assumes that the `mail` was removed.
+            {
+                // var mail = new ItemEnhanceMail(resultModel, 0, random.GenerateRandomGuid(), requiredBlockIndex);
+                // avatarState.Update(mail);
+            }
```
Comment (ipdae) [inline .Lib9c.Tests/Action/RapidCombinationTest.cs:678]: is it intended?

**Review by sonohoshi — APPROVED** [review]

File: .Lib9c.Tests/Action/RapidCombination6Test.cs (hunk 2 above)
Comment (boscohyun) [reply .Lib9c.Tests/Action/RapidCombination6Test.cs:678]: Yes, I hope someone without this knowledge would know that not adding mail was intentional.

File: .Lib9c.Tests/Action/RapidCombinationTest.cs (hunk 1 above)
Comment (boscohyun) [reply .Lib9c.Tests/Action/RapidCombinationTest.cs]: Removed!

File: .Lib9c.Tests/Action/RapidCombinationTest.cs (hunk 3 above)
Comment (boscohyun) [reply .Lib9c.Tests/Action/RapidCombinationTest.cs:677]: Yes, same.

**Review by ipdae — APPROVED** [review]

**Pull request merged**